src/Symfony/Component/HttpClient/CurlHttpClient.php

```diff
@@ -105,35 +105,29 @@ public function request(string $method, string $url, array $options = []): Respo
         $host = parse_url($authority, PHP_URL_HOST);
         $url = implode('', $url);
 
+        if (!isset($options['normalized_headers']['user-agent'])) {
+            $options['headers'][] = 'User-Agent: '.$options['normalized_headers']['user-agent'][] = 'Symfony HttpClient/Curl';
```

Comment:
I suggest splitting that in 2 statements to be more readable. An assignment inside a concatenation is quite easy to mis-read.